bundles/org.openhab.binding.tibber/src/test/java/org/openhab/binding/tibber/internal/TibberHandlerCallbackMock.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) 2010-2025 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.tibber.internal;
+
+import static org.junit.jupiter.api.Assertions.fail;
+import static org.mockito.Mockito.mock;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.config.core.ConfigDescription;
+import org.openhab.core.config.core.Configuration;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelGroupUID;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingStatusInfo;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.ThingHandlerCallback;
+import org.openhab.core.thing.binding.builder.ChannelBuilder;
+import org.openhab.core.thing.type.ChannelGroupTypeUID;
+import org.openhab.core.thing.type.ChannelTypeUID;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.openhab.core.types.TimeSeries;
+
+/**
+ * The {@link TibberHandlerCallbackMock} sets the callback for handler
+ *
+ * @author Bernd Weymann - Initial contribution
+ */
+@NonNullByDefault
+public class TibberHandlerCallbackMock implements ThingHandlerCallback {
+    ThingStatusInfo thingStatusInfo = new ThingStatusInfo(ThingStatus.UNKNOWN, ThingStatusDetail.NONE, null);
+
+    @Override
+    public void stateUpdated(ChannelUID channelUID, State state) {
+    }
+
+    @Override
+    public void postCommand(ChannelUID channelUID, Command command) {
+    }
+
+    @Override
+    public void sendTimeSeries(ChannelUID channelUID, TimeSeries timeSeries) {
+    }
+
+    @Override
+    public void statusUpdated(Thing thing, ThingStatusInfo thingStatus) {
+        synchronized (this) {
+            thingStatusInfo = thingStatus;
+            this.notifyAll();
+        }
+    }
+
+    public void waitFor(ThingStatus status) {
+        Instant stopWait = Instant.now().plus(10, ChronoUnit.SECONDS);
+        synchronized (this) {
+            while (!thingStatusInfo.getStatus().equals(status) && Instant.now().isBefore(stopWait)) {
```

Comment:
I wonder if you don't introduce a deadlock like situation as the above `statusUpdated()` also locks `this` and thet method is also called.
Why do you need this to be locked anyway?